File: MtgCardOrganizer.Core/Entities/Cards/CardInstance.cs
using System.Collections.Generic;
using MtgCardOrganizer.Core.Entities.Collections;

namespace MtgCardOrganizer.Core.Entities.Cards
{
    public class CardInstance : Entity
    {
        public bool Foil { get; set; }
        public bool Promo { get; set; }

        public int CardSetId { get; set; }
        public int CollectionId { get; set; }
        public CardSet CardSet { get; set; }
        public Collection Collection { get; set; }
    }
}

File: mtg-card-organizer/src/app/collection/models/card-instance.ts
import { CardSet } from '../../card/models/card-set';

export class CardInstance {
  id: number;

  foil = false;
  promo = false;

  cardSetId: number;
  cardSet: CardSet;

  public constructor(init?: Partial<CardInstance>) {
    Object.assign(this, init);
  }
}
